test/GeoSparqlHelpersTest.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022, University of Freiburg,
+// Chair of Algorithms and Data Structures
+// Authors: Hannah Bast <bast@cs.uni-freiburg.de
+
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <string>
+
+#include "../src/util/GeoSparqlHelpers.h"
+
+using ad_utility::wktDist;
+using ad_utility::wktLatitude;
+using ad_utility::wktLongitude;
+using ad_utility::detail::checkWktPoint;
+using ad_utility::detail::parseWktPoint;
+
+TEST(GeoSparqlHelpers, ParseWktPoint) {
+  // Test that the given WKT point parses correctly.
+  auto testParseWktPointCorrect = [](const std::string& point,
+                                     double expected_lng, double expected_lat) {
+    auto [lat, lng] = parseWktPoint(point);
+    ASSERT_FLOAT_EQ(expected_lng, lng);
+    ASSERT_FLOAT_EQ(expected_lat, lat);
+  };
+
+  // Test that the given WKT point is invalid.
+  auto testWktPointInvalid = [](const std::string& point) {
+    auto [lat, lng] = parseWktPoint(point);
+    ASSERT_TRUE(std::isnan(lng));
+    ASSERT_TRUE(std::isnan(lng));
+  };
+
+  // Some valid WKT points, including those from the test for `wktDist` below.
+  testParseWktPointCorrect("\"POINT(2.0 1.5)\"", 2.0, 1.5);
+  testParseWktPointCorrect("\"POINT(2.0 -1.5)\"", 2.0, -1.5);
+  testParseWktPointCorrect("\"PoInT(3   0.0)\"", 3.0, 0.0);
+  testParseWktPointCorrect("\"pOiNt(7 -0.0)\"", 7.0, 0.0);
+  testParseWktPointCorrect("\"pOiNt (  7  -0.0   )\"", 7.0, 0.0);
+  testParseWktPointCorrect("\"POINT(2.2945 48.8585)\"", 2.2945, 48.8585);
+  testParseWktPointCorrect("\"POINT(7.8529 47.9957)\"", 7.8529, 47.9957);
+
+  // All kinds of invalid WKT points (one of the quotes missing, one of the
+  // parentheses missing, explicit plus sign not allowed, scientific notation
+  // not allowed, it must be exactly two coordinates).
```

Comment:
See the comment about the grammar above. But I still made the split because it's easier to read with two blocks. Also added explicit tests that stuff like 42. and .42 is not allowed.